src/main/java/org/sonarlint/intellij/SonarLintIntelliJClient.kt

```diff
@@ -137,6 +142,57 @@ object SonarLintIntelliJClient : SonarLintRpcClientDelegate {
         suggestionsByConfigScopeId.forEach { (configScopeId, suggestions) -> suggestAutoBind(findProject(configScopeId), suggestions) }
     }
 
+    override fun suggestConnection(suggestionsByConfigScope: MutableMap<String, MutableList<ConnectionSuggestionDto>>?) {
+        if (suggestionsByConfigScope != null && suggestionsByConfigScope.size == 1) {
```

Comment:
You should actually iterate over all the map entries, not only the first one. The baby step we discussed about, it's for the list of `ConnectionSuggestionDto`, we can assume it should be only one element in the suggestion list, but we could have multiple config scope (if you have multiple projects opened for example)